Simple Incremental/Assets/Scripts/HelperAttack.cs

```diff
@@ -53,7 +53,7 @@ public void UntargetEnemy(GameObject go)
 
     public void OnEnemySpawn()
     {
-        if(target == null)
+        if(target == null || target.health <= 0)
```

Comment:
This is redundant. 

CharacterHealth is already sending a "Character Death" event that is handling this. Do not extend logic outside of the feature. This may cause conflict with other changes.

Reply:
So this was a bit of a temp patch so I could test the spawning was working. The attack helper is not properly targeting the next enemy because the target is not null. Especially if we are going to do pools and not delete we need to test if the enemy is alive or dead.

Reply:
The enemy manager provides a pool of enemies and removes them on death.

If you invoke an enemy spawned event and add a GameEventListener to the enemy manager listening for spawn events and adding the health, it should automatically be handled.

I suspect your scene did not have a GameEventListener on HelperAttack that was listening to Enemy Death and calling UntargetEnemy?